cypress/page/personal-form.page.ts

```diff
@@ -0,0 +1,105 @@
+import { forEach } from "cypress/types/lodash";
+
+class PersonalFormPage {
+    private nameInput: string;
+    private lastnameInput: string;
+    private emailInput: string;
+    private genderRadioBtn: string;
+    private dateOfBirthPicker: string;
+    private datePickerValues: string[];
+    private mobileNumberInput: string;
+    private hobbiesCheckbox: string;
+    private currentAddressInput: string;
+    private submitBtn: string;
+    private modalTitle: string;
+
+    constructor() {
+        this.nameInput = "#firstName";
+        this.lastnameInput = "#lastName";
+        this.emailInput = "#userEmail";
+        this.genderRadioBtn = '[name="gender"]';
+        this.dateOfBirthPicker = "#dateOfBirthInput";
+        this.datePickerValues = [".react-datepicker__year-select", ".react-datepicker__month-select", ".react-datepicker__day"];
+        this.mobileNumberInput = "#userNumber";
+        this.hobbiesCheckbox = '.custom-checkbox';
+        this.currentAddressInput = "#currentAddress";
+        this.submitBtn = "#submit";
+        this.modalTitle = "#example-modal-sizes-title-lg";
+    }
+
+    public fillForm(personalInformation: { name: string; lastName: string; email: string; gender: string; dateOfBirth: string; mobileNumber: string; hobbies: string[]; currentAddress: string;}): void {
+        cy.get(this.nameInput).type(personalInformation.name);
+        cy.get(this.lastnameInput).type(personalInformation.lastName);
+        cy.get(this.emailInput).type(personalInformation.email);
+        cy.get(this.genderRadioBtn).check(personalInformation.gender, {force: true});
+        this.fillDateInput(personalInformation.dateOfBirth);
```

Comment:
to fill the dateOfBirth you have two options:

- the long way, which you already implemented, using the date picker, if you want to continue using this way, you can convert the dateOfBirth in a date type instead of a string, in that way you can ease the implementation of this `fillDateInput` method. See reference of date type [here](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date)
- the short way, that is simulate that the user is writing the date:
   ```js
   cy.get(this.dateOfBirthPicker).type(`{selectAll}${userInformation.dateOfBirth}{enter}`)
   ```

I would like to see how it works with the long way, just because sometimes yo have no choice and you need to interact with datepickers.